scalr/resource_scalr_provider_configuration.go

```diff
@@ -677,7 +677,75 @@ func resourceScalrProviderConfigurationDelete(ctx context.Context, d *schema.Res
 	return nil
 }
 
-// changeParameters is used to change parameters for provider configuratio.
+func resourceScalrProviderConfigurationImport(ctx context.Context, d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
```

Comment:
The reason for resource import failure is that `Read()` function is not "pure" - it uses `d.Get()` to fetch values from the state (which is obviously missing in the case of importing the resource). I guess this is because API does not return values for sensitive fields and we do need them in the state. And having parameters in a nested block, it's impossible to just skip setting sensitive field, as we call `d.Set()` for the entire block. (cc @DayS1eeper is that true?)

Ideally, `Read()` functions should be able to refresh the resource with `d.Id()` **only**. 

Anyway I think custom importer here is an overkill, and this can be accomplished by tuning the `Read()` function a little.
Something like:

```
case "scalr":
	stateScalrParameters := d.Get("scalr").([]interface{})[0].(map[string]interface{})
	stateToken := stateScalrParameters["token"].(string)

	_ = d.Set("scalr", []map[string]interface{}{
		{
			"hostname": providerConfiguration.ScalrHostname,
			"token":    stateToken,
		},
	})
```
⬇️
```
case "scalr":
	var stateToken string
	if stateScalrParametersI, ok := d.GetOk("scalr"); ok {
		stateScalrParameters := stateScalrParametersI.([]interface{})
		if len(stateScalrParameters) > 0 {
			stateToken = stateScalrParameters[0].(map[string]interface{})["token"].(string)
		}
	}

	_ = d.Set("scalr", []map[string]interface{}{
		{
			"hostname": providerConfiguration.ScalrHostname,
			"token":    stateToken,
		},
	})
```

Reply:
++ I agree.